cmd/utils.go

```diff
@@ -464,14 +464,15 @@ func newCustomDialContext(dialTimeout, dialKeepAlive time.Duration) dialContext
 	}
 }
 
-func newCustomHTTPTransport(tlsConfig *tls.Config, dialTimeout, dialKeepAlive time.Duration) func() *http.Transport {
+func newCustomHTTPTransport(tlsConfig *tls.Config, dialTimeout time.Duration) func() *http.Transport {
 	// For more details about various values used here refer
 	// https://golang.org/pkg/net/http/#Transport documentation
 	tr := &http.Transport{
 		Proxy:                 http.ProxyFromEnvironment,
-		DialContext:           newCustomDialContext(dialTimeout, dialKeepAlive),
+		DialContext:           newCustomDialContext(dialTimeout, 15*time.Second),
 		MaxIdleConnsPerHost:   256,
-		IdleConnTimeout:       time.Minute,
+		MaxIdleConns:          256,
+		IdleConnTimeout:       5 * time.Minute,
```

Comment:
If I understood correctly, this means that if TCP keepalive sent each 15 seconds fails during 5 minutes, then prematurely close the connection in server's side. That means 20 consecutive failed retries (5*60/15). This does seem high for me.

Reply:
This is client side. If a keepalive check fails the connection will simply be closed and disregarded and no more checks will be done.

Reply:
`IdleConnTimeout` means keep connections alive up to 5 minutes after use in case another call is made to the same host. 

Only up to 256 connections will be kept alive as per above. Every 15 second the connection is checked. If this fails the connection is simply removed from the pool of idle connections.